src/main/java/Deadline.java

```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+    protected DateTime by;
+
+    public Deadline(String description, DateTime by) {
+        super(description);
+        this.by = by;
+        super.changeToFileFormat('D', description, by);
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + "(by: " + by + ")";
```

Comment:
Maybe you can consider using a String.format in this case. For example, _String.format("by: %s" ,by)_